include/boost/buffers/type_traits.hpp

```diff
@@ -32,6 +32,86 @@ struct is_const_buffer_sequence
     : std::integral_constant<bool, ...>{};
 #else
 
+template<class T, class = void>
+struct is_mutable_contiguous_container
+    : std::false_type
+{
+};
+
+template<class T>
+struct is_mutable_contiguous_container<
+    T,
+    void_t<
+        typename std::enable_if<
+            std::is_convertible<
+                decltype(std::declval<T&>().data()),
+                void*
+            >::value &&
+            std::is_convertible<
+                decltype(std::declval<T&>().size()),
+                std::size_t
+            >::value &&
+            !std::is_convertible<
+                T, mutable_buffer
+            >::value &&
+            !std::is_convertible<
+                T, const_buffer
+            >::value &&
+            std::is_trivial<
```

Comment:
It might be possible to relax this to `is_trivially_copyable` (doesn't make much difference in practice though).

Reply:
You know, that's a good point. The code currently uses `is_trivially_copyable` everywhere so this would be the outlier. The only tangible difference seems to be that `is_trivial` requires 0 user-defined constructors. I might be over-zealous here.